curriculum/challenges/english/25-front-end-development/lecture-the-var-keyword-and-hoisting/67335f45489c5a11b71d0ed5.md

```diff
@@ -8,7 +8,55 @@ dashedName: what-is-hoisting
 
 # --description--
 
-Watch the lecture video and answer the questions below.
+Watch the video or read the transcript and answer the questions below.
+
+# --transcript--
+
+What is hoisting?
+
+Hoisting is a behavior in JavaScript that often confuses beginners, but understanding it can help you avoid subtle bugs in your code. In simple terms hoisting is JavaScript's default behavior of moving declarations to the top of their respective scopes during the compilation phase before the code is executed.
+
+To understand hoisting, it's important to know that JavaScript runs in two phases: the compilation phase and the execution phase. During the compilation phase, the JavaScript engine goes through your code and sets up memory space for variables and functions. This is where hoisting comes into play.
+
+Let's start with variables hoisting, when you declare a variable using the `var` keyword, JavaScript hoists the declaration to the top of its scope. However it's crucial to note that only the declaration is hoisted, not the initialization. This means you can use a variable in your code before you have declared it, but its value will be `undefined` until you actually assign a value to it.
+
+```js
+console.log(x); // undefined
+var x = 5;
+console.log(5); // 5
+```
+
+In this code even though we use `x` before declaring it we don't get an error, instead we get `undefined`. This is because JavaScript hoists the declaration `var x` to the top of its scope but not the inialization `x = 5`. Its as if the code were rewritten like this:
```

Comment:
```suggestion
In this code even though we use `x` before declaring it we don't get an error, instead we get `undefined`. This is because JavaScript hoists the declaration `var x` to the top of its scope but not the inialization `x = 5`. It's as if the code were rewritten like this:
```